src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,53 @@
+import * as Icons from '@mui/icons-material';
+
+import { useTheme } from '@/hooks/useTheme';
+
+import { IconWrapper } from './Icon.style';
+
+/**
+ * @param [name] - 아이콘 이름을 컨벤션에 맞게끔 맨 뒤에 '-Rounded' 붙여주세요!
+ * @param [color] - 아이콘 색상을 지정해줄 수 있습니다.
+ * @param [fontSize] - 아이콘의 크기를 지정해줄 수 있습니다.
+ * @param [isShowBackground] - 아이콘의 배경을 보여줄지 말지 결정합니다. (ex - 라이트/다크모드 등)
+ * @param [onIconClick] - 마우스 클릭 핸들러 함수
+ */
+
+interface IconProps {
+  name: string;
+  color?: string;
+  fontSize?: number;
+  isShowBackground?: boolean;
+  onIconClick?: (event: React.MouseEventHandler<HTMLElement>) => void;
+}
+
+export default function Icon({
+  name,
+  color = '',
+  fontSize = 16,
+  isShowBackground = false,
+  onIconClick,
+  ...props
+}: IconProps) {
+  const theme = useTheme();
+  const IconComponent = Icons[name + 'Rounded'];
+
+  if (!IconComponent) {
+    console.error('아이콘 컴포넌트에 잘못된 이름을 입력하였습니다. : ', name);
+    return null;
+  }
```

Comment:
오.. 아이콘 이름 뒤에 Rounded suffix를 붙여서 접근하면
둥그런 아이콘들을 가져올 수 있게 되는군요! 신기하네요 ㅋㅋㅋ
 
없는 아이콘의 경우 예외 처리도 너무 좋습니다 👍
UI 상에서도 없는 아이콘이라는걸 바로 확인할 수 있으면 좋을 것 같은데
Empty Icon 같은걸 return 해주는건 어떨까요??

Reply:
엇..
vscode에서 돌려보니까 이런 에러가 발생하고 있어요!

`const IconComponent = Icons[name + 'Rounded'];`
요부분 Icons의 타입을 추론하지 못해서 string으로 접근하지 못하는 것 같기도 합니다..


```bash
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'typeof import("/Users/ssoobak/Desktop/Team-1e5i-AlgoBaro-FE/node_modules/@mui/icons-material/index")'.
  No index signature with a parameter of type 'string' was found on type 'typeof import("/Users/ssoobak/Desktop/Team-1e5i-AlgoBaro-FE/node_modules/@mui/icons-material/index")'.
```

Reply:
오.. 단순 console.error가 아닌 empty Icon 아이디어 너무 좋습니다~